packages/enclave-sdk/src/enclave-sdk.ts

```diff
@@ -173,6 +201,73 @@ export class EnclaveSDK {
     }
   }
 
+  /**
+   * Encrypt a vector using the configured protocol and generate a zk-SNARK proof using Greco
+   * @param data - The vector to encrypt
+   * @param publicKey - The public key to use for encryption
+   * @param circuit - The circuit to use for proof generation
+   * @returns The encrypted vector and the proof
+   */
+  public async encryptVectorAndGenProof(
```

Comment:
These two functions share most of the code. Worth considering to merge them somehow.